.github/workflows/continuous-integration.yaml

```diff
@@ -45,10 +45,38 @@ jobs:
         stack --no-terminal build                 --haddock --no-haddock-deps --flag ogmios:production
         stack --no-terminal test ogmios:unit      --haddock --no-haddock-deps --flag ogmios:production
 
+  docker:
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04 ]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: 📥 Checkout repository
+      uses: actions/checkout@v2.3.3
+
+    - name: 🧰 Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: 🐳 Login to DockerHub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_ACCESS_TOKEN }}
+
+    - name: 🔨 Build and push
+      uses: docker/build-push-action@v2
+      with:
+        context: server
+        push: true
+        tags: ktorz/ogmios:latest
+
   clients_TypeScript:
+    needs: [docker]
```

Comment:
Since the tests are ran on top of the docker stack, we need to build the latest version of image first. In the best case where we have already pre-built the cache, this only takes a couple of minutes.